scrapers/src/extractors/office.rs

```diff
@@ -0,0 +1,321 @@
+mod tests {
+    use super::*;
+
+    #[test]
+    fn extract_meta() {
+        let tests: Vec<(&'static str, Option<&'static str>)> = vec![
+            ("United States", None),
+            ("U.S.", None),
+            ("State", None),
+            ("Senator", None),
+            ("House", None),
+            ("Representative", None),
+            ("District", None),
+            ("County", None),
+            ("Board", None),
+            ("Education", None),
+            ("Regent", None),
+            ("University", None),
+            ("Attorney", None),
+            ("Court", None),
+            ("Judge", None),
+            ("Appeals", None),
+            ("Transportation", None),
+            ("Director", None),
+            // ----
+            ("U.S. Senate", Some("U.S. Senate")),
+            ("U.S. Senator", Some("U.S. Senate")),
+            ("United States Senate", Some("U.S. Senate")),
+            ("United States Senator", Some("U.S. Senate")),
+            ("U.S. House", Some("U.S. House")),
+            ("U.S. Representative", Some("U.S. House")),
+            ("United States House", Some("U.S. House")),
+            ("United States Representative", Some("U.S. House")),
+            ("State Senate", Some("State Senate")),
+            ("State Senator", Some("State Senate")),
+            ("State House", Some("State House")),
+            ("State Representative", Some("State House")),
+            ("Board of Education", Some("Board of Education")),
+            ("Board of Regents", Some("Board of Regents")),
+            ("Regent of the University", Some("Board of Regents")),
+            ("District Attorney", Some("District Attorney")),
+            ("District Court Judge", Some("District Court Judge")),
+            ("County Court Judge", Some("County Court Judge")),
+            ("Court of Appeals Judge", Some("Court of Appeals Judge")),
+            ("Supreme Court Justice", Some("Supreme Court Justice")),
+            (
+                "Regional Transportation District Director",
+                Some("Regional Transportation District Director"),
+            ),
+        ];
+
+        for (input, expected) in tests {
+            assert_eq!(
+                extract_office_meta(input)
+                    .as_ref()
+                    .map(|meta| meta.name.as_str()),
+                expected,
+                "\n\n  Test Case: '{input}'\n"
+            );
+        }
+    }
+
+    #[test]
+    fn extract_qualifier() {
+        let tests: Vec<(&'static str, Option<&'static str>)> = vec![
+            ("District", None),
+            ("District ", None),
+            ("District Attorney", None),
+            ("District 0", Some("0")),
+            ("Something District 1", Some("1")),
+            ("District 12", Some("12")),
+            ("District A", Some("A")),
+            ("District AB", Some("AB")),
+            ("District 1A", Some("1A")),
+            (" District 01 ", Some("01")), // TODO weird edge case?
```

Comment:
For the MN data we've been normalizing these cases "01" -> "1"

Reply:
ah, yeah that makes sense, will do